### ISSUE TITLE
[1.5] Class "craft\commerce\fields\Products" not found

### ISSUE THREAD

**Issue by Mosnar** [verbb/formie#677]
**Description**

I just got this error while creating a Feed Me feed.

I'm running the latest commit to get around the issue mentioned in #676 

I do not have Commerce installed.

Full stack trace:

```
Error: Class "craft\commerce\fields\Products" not found in /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/verbb/formie/src/fields/formfields/Products.php:27
Stack trace:
#0 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/composer/ClassLoader.php(571): include()
#1 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile()
#2 [internal function]: Composer\Autoload\ClassLoader->loadClass()
#3 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/feed-me/src/services/Fields.php(75): class_exists()
#4 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\feedme\services\Fields->init()
#5 [internal function]: yii\base\BaseObject->__construct()
#6 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/di/Container.php(412): ReflectionClass->newInstanceArgs()
#7 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/di/Container.php(171): yii\di\Container->build()
#8 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/BaseYii.php(345): yii\di\Container->get()
#9 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/di/ServiceLocator.php(137): yii\BaseYii::createObject()
#10 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/di/ServiceLocator.php(74): yii\di\ServiceLocator->get()
#11 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Extension/CoreExtension.php(1431): yii\di\ServiceLocator->__get()
#12 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/cms/src/helpers/Template.php(106): twig_get_attribute()
#13 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/e3/e3b56f444a89baa677cfdf4db68169677e08612ae20967dd38a01bf34e354fe4.php(226): craft\helpers\Template::attribute()
#14 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(405): __TwigTemplate_2c04fd217cb2d95cedc778955e5c373db5c416dde8f84d6ab4567fd774a7ec0f->doDisplay()
#15 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#16 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/38/38e89337a5a3dfac4a047188afd5b1b23dfe0fbe64a2fa3ec426c84b9d2701e9.php(169): Twig\Template->display()
#17 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(182): __TwigTemplate_38a3ee001366b0d3bc397a06538fb9f90e44684568420ab6999d4c716868c8a9->block_content()
#18 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/8f/8f9585b4c8ca4408ef7be3a66f112221047f8272d813fb5fad6adb97962c074c.php(535): Twig\Template->displayBlock()
#19 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(182): __TwigTemplate_c9772006eb2d0862f9b3fff9899eaf3be940a44f725f68cf27ce8c800e2f60fc->block_main()
#20 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/8f/8f9585b4c8ca4408ef7be3a66f112221047f8272d813fb5fad6adb97962c074c.php(346): Twig\Template->displayBlock()
#21 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(182): __TwigTemplate_c9772006eb2d0862f9b3fff9899eaf3be940a44f725f68cf27ce8c800e2f60fc->block_body()
#22 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/c0/c030acb040c591ee0bcee477a8a30ae8979cd1ebde7ccb37f92eb62fbf6d6e6c.php(96): Twig\Template->displayBlock()
#23 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(405): __TwigTemplate_01fd67683c384489efeb51a27a5ba64a544bbc8d6901f5313165ee0123dc7121->doDisplay()
#24 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#25 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/bb/bbc8928225ebb01f0f4d05c86f10ea1a5dc6a197400a6508e72bb42d03657b97.php(56): Twig\Template->display()
#26 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(405): __TwigTemplate_8c314d4e276ac4abc83af085d00a0b13e39175c420de55b7cba8fa146b5a34fb->doDisplay()
#27 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#28 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/8f/8f9585b4c8ca4408ef7be3a66f112221047f8272d813fb5fad6adb97962c074c.php(165): Twig\Template->display()
#29 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(405): __TwigTemplate_c9772006eb2d0862f9b3fff9899eaf3be940a44f725f68cf27ce8c800e2f60fc->doDisplay()
#30 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#31 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/e8/e851612228fb4ca7895f47a9f2e13dd8d584ca0c1b10f1a65a60e98aadec2950.php(57): Twig\Template->display()
#32 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(405): __TwigTemplate_986206168cc3afc0788ea3fa4007a7f4b35488b2bb96e98ad8ace703481432e7->doDisplay()
#33 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#34 /home/ploi/redacted-deploy/redacted/07122021_222307/storage/runtime/compiled_templates/38/38e89337a5a3dfac4a047188afd5b1b23dfe0fbe64a2fa3ec426c84b9d2701e9.php(79): Twig\Template->display()
#35 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(405): __TwigTemplate_38a3ee001366b0d3bc397a06538fb9f90e44684568420ab6999d4c716868c8a9->doDisplay()
#36 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling()
#37 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Template.php(390): Twig\Template->display()
#38 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render()
#39 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render()
#40 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/cms/src/web/View.php(408): Twig\Environment->render()
#41 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/cms/src/web/View.php(461): craft\web\View->renderTemplate()
#42 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/cms/src/web/Controller.php(201): craft\web\View->renderPageTemplate()
#43 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/feed-me/src/controllers/FeedsController.php(104): craft\web\Controller->renderTemplate()
#44 [internal function]: craft\feedme\controllers\FeedsController->actionMapFeed()
#45 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#46 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams()
#47 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/base/Module.php(534): yii\base\Controller->runAction()
#48 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/cms/src/web/Application.php(287): yii\base\Module->runAction()
#49 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/web/Application.php(104): craft\web\Application->runAction()
#50 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/craftcms/cms/src/web/Application.php(272): yii\web\Application->handleRequest()
#51 /home/ploi/redacted-deploy/redacted/07122021_222307/vendor/yiisoft/yii2/base/Application.php(392): craft\web\Application->handleRequest()
#52 /home/ploi/redacted-deploy/redacted/07122021_222307/web/index.php(22): yii\base\Application->run()
#53 {main}
```

**Comment by engram-design** [comment]
Just trying to replicate this, at what stage of the process in Feed Me does this get raised?

EDIT: Ah, the final mapping step!

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.0",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.1](https://github.com/verbb/formie/releases/tag/1.5.1)
